stanford_soe_r25.feed.inc

```diff
@@ -0,0 +1,203 @@
+<?php
+
+// takes ajax requests from FullCalendar, passes them on to the 25Live web api, and returns the results as JSON
+
+// bit of DRY code to return values from the web services api xml
+function _stanford_soe_r25_feed_get_value(&$results, $name, $key) {
+  $return_val = '';
+  if (isset($results['vals'][$results['index'][$name][$key]]['value'])) {
+    $return_val = $results['vals'][$results['index'][$name][$key]]['value'];
+  }
+  return $return_val;
+}
+
+// This is the ajax function called by FullCalendar with a room id, start date and end date in the POST data.
+// Returns JSON array of event data to FullCalendar
+function _stanford_soe_r25_feed($get_room_id = NULL) {
+
+  // format the request to the 25Live API from either POST or GET arrays
+  $rooms = _stanford_soe_r25_room_config_load();
+  $room_id = '';
+  $space_id = NULL;
+  $start = '';
+  $end = '';
+  if (!empty($_POST['room_id']) && !empty($rooms[$_POST['room_id']]['space_id'])) {
+    $room_id = $_POST['room_id'];
+    $space_id = $rooms[$_POST['room_id']]['space_id'];
+    $start = str_replace('-', '', $_POST['start']);
+    $end = str_replace('-', '', $_POST['end']);
+  }
+  else {
+    if (empty($get_room_id) || empty($rooms[$get_room_id]['space_id'])) {
+      drupal_not_found();
+    }
+    else {
+      $room_id = $get_room_id;
+      $space_id = $rooms[$get_room_id]['space_id'];
+      if (!empty($_GET['start'])) {
+        $start = str_replace('-', '', $_GET['start']);
+      }
+      if (!empty($_GET['end'])) {
+        $end = str_replace('-', '', $_GET['end']);
+      }
+    }
+  }
+
+  // depending on the logged in user requesting this information, we want to include
+  // links to contact the event scheduler, or to confirm or cancel the event
+  $approver_list = array();
+  if (!empty($rooms[$room_id]['approver_secgroup_id'])) {
+    $approver_list = _stanford_soe_r25_security_group_emails($rooms[$room_id]['approver_secgroup_id']);
+  }
+  $approver = FALSE;
+
+  // if the user is Drupal user 1 or can administer rooms, let them approve
+  // and cancel events
+  global $user;
+  if ($user->uid == 1 || user_access('administer stanford r25') ||
+    ($user->uid > 0 && in_array($user->mail, $approver_list))
```

Comment:
wouldnt only users with a email have a uid? not sure the uid check is doing anything